lib/systemfile.g

```diff
@@ -0,0 +1,27 @@
+(function()
+    local deps, i, standalone, initfiles, entry;
+
+    # Add JuliaInterface to the needed GAP packages, such that it is loaded
+    # early enough for being required by other GAP packages,
+    # and such that it is available when user files get read
+    # via the GAP command line.
+    deps:= SHALLOW_COPY_OBJ( GAPInfo.Dependencies.NeededOtherPackages );
+    APPEND_LIST_INTR( deps, [ [ "JuliaInterface", ">= 0.4.4" ] ] );
+    GAPInfo.Dependencies:= MakeImmutable( rec( NeededOtherPackages:= deps ) );
+
+    # Notify that GAP.jl initiates the loading process.
+    for i in [ 1 .. LEN_LIST( GAPInfo.SystemCommandLine )-1 ] do
+      if GAPInfo.SystemCommandLine[i] = "-c" then
+        if POSITION_SUBSTRING( GAPInfo.SystemCommandLine[i+1], "__JULIAINTERNAL_LOADED_FROM_JULIA_STAND_ALONE\", true", 0 ) <> fail then
+          standalone:= true;
+        elif POSITION_SUBSTRING( GAPInfo.SystemCommandLine[i+1], "__JULIAINTERNAL_LOADED_FROM_JULIA_STAND_ALONE\", false", 0 ) <> fail then
+          standalone:= false;
+        fi;
+      fi;
+    od;
+    if IsBound( standalone ) then
+      # Set the flag that notifies the 'JuliaInterface' package:
+      # GAP.jl has initiated to load GAP.
+      BIND_GLOBAL( "__JULIAINTERNAL_LOADED_FROM_JULIA", true );
+    fi;
```

Comment:
All this just to set a GAP variable. I see three alternatives to this:
1. have two different `systemfile.g`, one for standalone mode, one for non-standalone ("embedded") mode
2. only in standalone mode, inject code just before the start of `init.g` which set `__JULIAINTERNAL_LOADED_FROM_JULIA` to `true`; two ways to achieve this come to mind:
    1. provide our own `init.g` and ensure it is visible in a GAPROOT before the one with the real `init.g`. Then have that custom `init.g` set `__JULIAINTERNAL_LOADED_FROM_JULIA` and read the "real" `init.g`.
    2. set the C kernel variable `SyLoadSystemInitFile` to 0 (zero). That ensures that `GAP_Initialize` returns *before* `init.g` is read (but well after the kernel itself is set up, so we can e.g. assign to GAP global variables). It also means that we then need to issue the equivalent of `READ("lib/init.g")`.

Reply:
OK, I missed your comments on issue #518, otherwise my comment above would be a bit different :-). Let me first say that I think the idea of adding `--systemfile` is the right call, and I fully support it. I just think it's not mutually exclusive with inserting some code before reading `init.g`. I think the two would complement each other quite nicely. I might hack up the `SyLoadSystemInitFile` idea just to try it ...

In fact, I am actually think about combining the `SyLoadSystemInitFile` approach with the switch to GAP_jll anyway: the idea being that we move the definition of `GapObj` as well as the call to `GAP_Initialize` into `GAP_jll`, but with `SyLoadSystemInitFile` set to 0, so that it does not process `init.g`. 

By doing this, when `GAP.jl` is loaded, `ForeignPtr.MPtr` should already be defined, and it might be possible for us to do some things that are not possible right now, or at least do them in a somewhat cleaner fashion. Like writing things such as `const T_MACFLOAT = ValueGlobalVariable(:T_MACFLOAT)` should become possible at the top level of the package. I guess in a sense this would bring back the old `GAPTypes` package, at least in spirit, but also way beyond it.  (However, I need to think about whether and how this can be made to work while retaining support for loading GAP first, then loading Julia from GAP).